boefjes/boefjes/plugins/kat_withaname/normalize.py

```diff
@@ -0,0 +1,72 @@
+import json
+from collections.abc import Iterable
+
+import structlog
+
+from octopoes.models import OOI, Reference
+from octopoes.models.ooi.findings import Finding, KATFindingType
+
+logger = structlog.getLogger(__name__)
+
+
+def run(input_ooi: dict[str, str], raw: bytes) -> Iterable[OOI]:
+    """Normalize witha.name output."""
+    result = json.loads(raw)
+
+    if result["targets"] is None:
+        logger.info("No targets found in witha.name output")
+        return
+
+    input_ooi_reference = Reference.from_str(input_ooi["primary_key"])
+    found_targets = None
+
+    ooi_category = "IP" if input_ooi["primary_key"].startswith("IPAddress") else "Hostname"
```

Comment:
I believe the entire object is serialized in the `input_ooi` object, so you can conveniently get the object type like this:
`input_ooi["object_type"]`

Reply:
Fixed in 881e0c5

Normalizer now checks if it is dealing with a hostname, and if not; it will assume it is an IP